docker/ray-ml/install-ml-docker-requirements.sh

```diff
@@ -32,10 +34,13 @@ pip --no-cache-dir install -U \
            -r ray-docker-requirements.txt
 
 # explicitly install (overwrite) pytorch with CUDA support
-pip --no-cache-dir install -U \
+# do not use compiled requirements as constraint here as some
+# device-specific constraints can't be met (e.g. torch+cpu).
+# Instead, we don't use `-U` so other dependencies don't get updated.
+pip --no-cache-dir install \
```

Comment:
could you explain why "device-specific constraints" cannot be met? do we plan to meet them in the future?

Reply:
torch pins to e.g. `torch==2.0.1+cpu`, which conflicts if we request e.g. `torch==2.0.1+cu118` (it also conflicts with just `torch==2.0.1`. We can only compile with one of these constraints and currently use the CPU version (`dl-cpu-requirements.txt`). 

Like mentioned in my comment, for the ml images we can replace the device suffixes. At the moment we just don't use the constraints which will update the dependencies in the file but no other ones (since `-U` is not used)

Reply:
It's probably cleaner to replace the device suffixes, so I'll do that in the next update